src/lib.rs

```diff
@@ -136,15 +136,14 @@ impl<'a, NetVal: NetworkValidation, T> Uri<'a, NetVal, T> {
 
 impl<'a, T> Uri<'a, bitcoin::address::NetworkUnchecked, T> {
     /// Checks that the bitcoin network in the URI is `network`.
-    pub fn require_network(self, network: bitcoin::Network) -> Result<Uri<'a, bitcoin::address::NetworkChecked, T>, InvalidNetworkError> {
-        if self.address.is_valid_for_network(network) {
-            Ok(self.assume_checked())
-        } else {
-            Err(InvalidNetworkError {
-                required: network,
-                found: *self.address.network(),
-            })
-        }
+    pub fn require_network(self, network: bitcoin::Network) -> Result<Uri<'a, bitcoin::address::NetworkChecked, T>, bitcoin::address::ParseError> {
+        Ok(Uri {
+            address: self.address.require_network(network)?,
+            amount: self.amount,
+            label: self.label,
+            message: self.message,
+            extras: self.extras,
+        })
```

Comment:
FYI I'm not sure why exactly I made my own type here but now it looks better since the resulting error is convertible to `UriError`.

Reply:
it might make sense to actually return https://docs.rs/bitcoin/latest/bitcoin/address/struct.NetworkValidationError.html instead of the general ParseError

Reply:
Oh, yes, we should do that but also impl `From<NetworkValidationError> for UriError`.